app/preload/http.ts

```diff
@@ -44,13 +48,39 @@ async function httpsGet<T>(
 }
 
 /**
- * Loads the newest transactions on the given address, using the given filters.
- * @param transactionFilter is used to filter the request, however only from/to date, and filters for reward types are currently used.
+ * Filters transactions on their type. This extra filtering is required
+ * as the wallet proxy does not support a fine grained filtering on
+ * types at this time.
+ * @param transactionFilter the filtering to apply to the transactions
  */
-async function getNewestTransactions(
+function filterTransactionsOnType(
+    transactionFilter: TransactionFilter,
+    transactions: IncomingTransaction[]
+) {
+    const allowedTypes = getActiveBooleanFilters(transactionFilter);
+    const filteredTransactions = transactions.filter((t) =>
+        allowedTypes.includes(t.details.type)
+    );
+    return filteredTransactions;
+}
+
+/**
+ * Loads transactions from the wallet proxy for the account with the provided address. The supplied
+ * transaction filter is applied for reward filtering, and filtering based on dates. All other parts
+ * of the filter is ignored, as the wallet proxy does not support it!
+ * @param address the account address to get transactions for
+ * @param transactionFilter is used to filter the request, however only from/to date, and filters for reward types are currently used
```

Comment:
This line is somewhat incorrect, because we filter on the remaining types afterwards.